### PR TITLE
Changes Required for ROOT v6.28

### PR DESCRIPTION
**Pull request by jeanyvesb9**
Changes required to run HistFitter on ROOT v6.26. This addresses #146.

- Fixed RooList dependency: removed custom implementation of RooHist and RooPlot; replaced the ratio hist method with a dedicated function in Utils.cxx
- Removed the NormFactor 'Const' parameter in XMLs and each Python's sample config.
- Fixed other library dependencies: RooNLLVar.h, TMath.h
- Now requiring at least ROOT 6.28

### CODE REVIEW

**Comment by jeanyvesb9** [comment]
Ok! This should now be ready to be merged @kratsg @longjon929

**Comment by longjon929** [comment]
Thanks, I'll try to take a look tomorrow.

File: src/RooHist.cxx
Comment (longjon929) [inline src/RooHist.cxx:1]: The only modifications here and to RooPlot were for the ratio plot?

**Comment by longjon929** [comment]
Thanks for preparing this.  I didn't spot anything other than a minor point in the README.

File: README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ git add HistFitter && git commit -m "Add HistFitter submodule"
 
 ### Recommended Root version
 
-Recommended Root version is 6.26.08.  This version of HistFitter is not compatible with ROOT versions < 6.26.
+Recommended Root version is 6.28. This version of HistFitter is not compatible with ROOT versions < 6.28.
```
[the remark this answers is not in the document]
Comment (jeanyvesb9) [reply README.md]: Done!

File: python/sample.py [hunk 2]
```diff
@@ -1514,7 +1512,7 @@ def __str__(self):
         
         if len(self.normFactor)>0:
             for normFactor in self.normFactor:
-                self.sampleString += f"    <NormFactor Name=\"{normFactor[0]}\" Val=\"{normFactor[1]:g}\" High=\"{normFactor[2]:g}\" Low=\"{normFactor[3]:g}\" Const=\"{normFactor[4]}\" />\n"
```
[the remark this answers is not in the document]
Comment (jeanyvesb9) [reply python/sample.py:1517]: Yes, it's mentioned in the commit where the "const" option was removed (https://github.com/root-project/root/commit/40fd04fe74c08607c47da5c5b81da88dd5f2850a) that the use of that argument has been deprecated for a very long time.

File: src/RooHist.cxx
Comment (jeanyvesb9) [reply src/RooHist.cxx:1]: Yes. All the other code was the same as in the old RooPlot and RooHist implementations (before the RooList removal)

**Pull request merged**